### ISSUE TITLE
DOCKER_VERNEMQ_DISCOVERY_KUBERNETES=1 results in startup errors

### ISSUE THREAD

**Issue by matthastie** [vernemq/docker-vernemq#159]
We experience these two errors when we start VerneMQ:

```
2019-09-10T17:34:38.242986488Z /usr/sbin/start_vernemq: line 53: [: too many arguments
2019-09-10T17:34:38.244039701Z sed: -e expression #1, char 56: unknown option to `s'
```

This appears to be a result of recent commits to the following lines of `vernemq.sh`.

```
52: VERNEMQ_KUBERNETES_SUBDOMAIN=${DOCKER_VERNEMQ_KUBERNETES_SUBDOMAIN:-(curl -X GET $insecure --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt https://kubernetes.default.svc.$DOCKER_VERNEMQ_KUBERNETES_CLUSTER_NAME/api/v1/namespaces/$DOCKER_VERNEMQ_KUBERNETES_NAMESPACE/pods?labelSelector=$DOCKER_VERNEMQ_KUBERNETES_LABEL_SELECTOR -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" | jq '.items[0].spec.subdomain' | sed 's/"//g' | tr '\n' '\0')}
53: if [ $VERNEMQ_KUBERNETES_SUBDOMAIN == "null" ]; then
54: VERNEMQ_KUBERNETES_HOSTNAME=${MY_POD_NAME}.${DOCKER_VERNEMQ_KUBERNETES_NAMESPACE}.svc.${DOCKER_VERNEMQ_KUBERNETES_CLUSTER_NAME}
55: else
56: VERNEMQ_KUBERNETES_HOSTNAME=${MY_POD_NAME}.${VERNEMQ_KUBERNETES_SUBDOMAIN}.${DOCKER_VERNEMQ_KUBERNETES_NAMESPACE}.svc.${DOCKER_VERNEMQ_KUBERNETES_CLUSTER_NAME}
    fi
```

We suspect that the curl stats emitted from line 52 might be including space characters in the output, which causes line 53 to fail, due to missing quotes around the dereference of the lhs. If it is indeed the stats that are problematic, adding -s (silent) to curl can give a hostname output, but we conjecture that even if this were to be corrected, the generated hostname (with subdomain calculated) would not be correct for our environment. We request that you please resurrect commit 4c78483 to provide users an option to explicitly configure the subdomain value.

```
echo $(curl -q -X GET --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt https://kubernetes.default.svc.cluster.local/api/v1/namespaces/mqtt/pods?labelSelector=app=vernemq -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" | jq '.items[].spec.hostname' | sed -e 's/"//g' | tr '\n' ' ')
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 16242    0 16242    0     0   229k      0 --:--:-- --:--:-- --:--:--  229k
vernemq-0
echo $(curl -s -X GET --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt https://kubernetes.default.svc.cluster.local/api/v1/namespaces/mqtt/pods?labelSelector=app=vernemq -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" | jq '.items[].spec.hostname' | sed -e 's/"//g' | tr '\n' ' ')
vernemq-0
$ 
```

The output above should demonstrate that our default service account has adequate privileges, and our environment is presented below for completeness:

```
        image: erlio/docker-vernemq:1.9.2
        - name: MY_POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: DOCKER_VERNEMQ_ALLOW_ANONYMOUS
          value: "off"
        - name: DOCKER_VERNEMQ_DISCOVERY_KUBERNETES
          value: "1"
        - name: DOCKER_VERNEMQ_KUBERNETES_LABEL_SELECTOR
          value: app=vernemq
```

**Comment by larshesel** [comment]
Hi, thanks for opening the issue.

Isn't it possible to set `DOCKER_VERNEMQ_KUBERNETES_SUBDOMAIN` now in which case the `curl` command won't be executed?

Would you be able to provide a PR fixing the issue or reverting the problematic commit?

**Comment by matthastie** [comment]
I was able to confirm: 

* commit 077a4bf02d5 breaks curl execution, but
* explicit set of DOCKER_VERNEMQ_KUBERNETES_SUBDOMAIN is a possible workaround.

- VERNEMQ_KUBERNETES_SUBDOMAIN=${DOCKER_VERNEMQ_KUBERNETES_SUBDOMAIN:(curl
+ VERNEMQ_KUBERNETES_SUBDOMAIN=${DOCKER_VERNEMQ_KUBERNETES_SUBDOMAIN:-(curl

We are able to workaround this with the explicit specification of subdomain.

**Comment by mdrobny** [comment]
I have the same problem, currently using newest version of Chart with `replicaCount` > 1 doesn't work.

This PR https://github.com/vernemq/docker-vernemq/pull/154 broke Chart so maybe @samk-mesh do you know how to fix it? It worked properly without `DOCKER_VERNEMQ_KUBERNETES_SUBDOMAIN`

**Comment by larshesel** [comment]
Hi all @c-meier just created a PR for this issue in #160 2ould you be able to test this out/review it @matthastie @mdrobny

**Comment by larshesel** [comment]
The fix in #160 was merged. Thanks @c-meier for the contribution!
